samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -363,15 +364,22 @@ File getStoreLocation(String storeName) {
 
     // Step 3
     metadata.forEach((systemStream, systemStreamMetadata) -> {
+
         // get the partition metadata for each system stream
         Map<Partition, SystemStreamMetadata.SystemStreamPartitionMetadata> partitionMetadata =
           systemStreamMetadata.getSystemStreamPartitionMetadata();
 
         // For SSPs belonging to the system stream, use the partition metadata to get the oldest offset
-        Map<SystemStreamPartition, String> offsets = systemStreamToSsp.get(systemStream).stream()
-          .collect(
-              Collectors.toMap(Function.identity(), ssp -> partitionMetadata.get(ssp.getPartition()).getOldestOffset()));
-
+        // if partitionMetadata was not obtained for any SSP, populate oldest-offset as null
+        // Because of https://bugs.openjdk.java.net/browse/JDK-8148463 using lambda will NPE when getOldestOffset() is null
+        Map<SystemStreamPartition, String> offsets = new HashMap<>();
+        for (SystemStreamPartition ssp : systemStreamToSsp.get(systemStream)) {
+          if (partitionMetadata == null || partitionMetadata.get(ssp.getPartition()) == null) {
```

Comment:
SystemStreamMetadata#getSystemStreamPartitionMetadata is also used in CoordinatorStreamStore, ContainerStorageManager and other places, we should also do null handling there right?

Isn't it better to do at StreamMetadataCache level?

Reply:
the null values are on oldestOffset in the returned metadata-object not the returned objectl itself (returned val is by us and is not-null it appears).